diff --git a/CMakeLists.txt b/CMakeLists.txt
index 570af71..58539b2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -297,7 +297,7 @@ add_definitions(-DHIPRT_PUBLIC_REPO)
 
 
 # Enable CUDA if possible
-include(${CMAKE_CURRENT_SOURCE_DIR}/contrib/Orochi/Orochi/enable_cuew.cmake)
+# include(${CMAKE_CURRENT_SOURCE_DIR}/contrib/Orochi/Orochi/enable_cuew.cmake)
 
 
 
@@ -363,16 +363,16 @@ endif()
 
 # precompile kernels:
 if(PRECOMPILE)
-	message(">> Execute: ${PYTHON_EXECUTABLE} compile.py --nvidia --hipSdkPath \"${HIP_FINAL_PATH}\"")
+	message(">> Execute: ${PYTHON_EXECUTABLE} compile.py --hipSdkPath \"${HIP_FINAL_PATH}\"")
 	execute_process(
-		COMMAND ${PYTHON_EXECUTABLE} compile.py --nvidia --hipSdkPath ${HIP_FINAL_PATH}
+		COMMAND ${PYTHON_EXECUTABLE} compile.py --hipSdkPath ${HIP_FINAL_PATH}
 		WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/scripts/bitcodes
 		)
 	
 	if(NOT NO_UNITTEST)
-		message(">> Execute: ${PYTHON_EXECUTABLE} precompile_bitcode.py --nvidia --hipSdkPath \"${HIP_FINAL_PATH}\"")
+		message(">> Execute: ${PYTHON_EXECUTABLE} precompile_bitcode.py --hipSdkPath \"${HIP_FINAL_PATH}\"")
 		execute_process(
-			COMMAND ${PYTHON_EXECUTABLE} precompile_bitcode.py --nvidia --hipSdkPath ${HIP_FINAL_PATH}
+			COMMAND ${PYTHON_EXECUTABLE} precompile_bitcode.py --hipSdkPath ${HIP_FINAL_PATH}
 			WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/scripts/bitcodes
 			)
 	endif()
