diff --git a/manifoldDeps.cmake b/manifoldDeps.cmake
index 182145b..47e6cb5 100644
--- a/manifoldDeps.cmake
+++ b/manifoldDeps.cmake
@@ -1,8 +1,8 @@
 include(FetchContent)
 include(GNUInstallDirs)
-find_package(PkgConfig)
+find_package(PkgConfig QUIET)
 if(MANIFOLD_PAR STREQUAL "TBB")
-    find_package(TBB)
+    find_package(TBB QUIET)
 endif()
 if (PKG_CONFIG_FOUND)
     pkg_check_modules(Clipper2 Clipper2)
@@ -37,15 +37,15 @@ else()
     endif()
 endif()
 
-FetchContent_Declare(glm
-    GIT_REPOSITORY https://github.com/g-truc/glm.git
-    GIT_TAG b06b775c1c80af51a1183c0e167f9de3b2351a79
-    GIT_PROGRESS TRUE
-    FIND_PACKAGE_ARGS NAMES glm
-)
-FetchContent_MakeAvailable(glm)
-
+find_package(glm QUIET)
 if(NOT glm_FOUND)
+    set(GLM_BUILD_INSTALL "ON" CACHE STRING "install glm")
+    FetchContent_Declare(glm
+        GIT_REPOSITORY https://github.com/g-truc/glm.git
+        GIT_TAG b06b775c1c80af51a1183c0e167f9de3b2351a79
+        GIT_PROGRESS TRUE
+    )
+    FetchContent_MakeAvailable(glm)
     message(STATUS "glm not found, downloading from source")
     if(NOT EMSCRIPTEN)
         install(TARGETS glm EXPORT glmTargets)
diff --git a/src/utilities/CMakeLists.txt b/src/utilities/CMakeLists.txt
index bdd2253..46c463a 100644
--- a/src/utilities/CMakeLists.txt
+++ b/src/utilities/CMakeLists.txt
@@ -60,7 +60,7 @@ endif()
 target_include_directories(${PROJECT_NAME} INTERFACE
     $<INSTALL_INTERFACE:include/${CMAKE_PROJECT_NAME}>
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>)
-target_link_libraries(${PROJECT_NAME} INTERFACE glm::glm)
+target_link_libraries(${PROJECT_NAME} INTERFACE $<BUILD_INTERFACE:glm::glm>)
 
 if(NOT DEFINED thrust_SOURCE_DIR)
     set(thrust_SOURCE_DIR ${_THRUST_INCLUDE_DIR})
0 replies


